docker/Dockerfile

```diff
@@ -91,19 +91,13 @@ RUN python -m virtualenv $VIRTUAL_ENV
 ENV PATH "$VIRTUAL_ENV/bin:$PATH"
 
 # Prevent pip from complaining about available upgrades inside virtualenv
-RUN pip install --upgrade pip setuptools wheel
+RUN pip install --upgrade pip setuptools wheel && \
+  rm -r $HOME/.cache/pip
 
 # We need a MuJoCo key to install mujoco_py
 # In this step only the presence of the file mjkey.txt is required, so we only
 # create an empty file
 RUN touch $HOME/.mujoco/mjkey.txt
-RUN pip install garage && \
```

Comment:
where will garage be installed in the container now?

Reply:
doesn't this make this container useless as a standalone?

Reply:
Okay so my plan was to fix this for now so that CI becomes faster and then add a separate target that installs garage from pypi

Reply:
@avnishn it's installed at the end of Dockerfile, using setup.py

Reply:
okay -- can you do that in this PR?

we should probably make that the default target, something like `rlworkgroup/garage` where `rlworkgroup/garage:2020.06.0` corresponds to the releases, etc.

Reply:
This was a problem when our latest release on pypi was 2019.10 because the deps were different. So docker would end up installing both, say TF 1.15 and TF 2, which are like 500 MB each at download and much bigger at install. Now our pypi release and master have almost same deps, eventually it'll drift again.

Reply:
I agree. we will need some solution to this, but not right now.

Perhaps file an issue about version-specific Dockerfile?

Reply:
The solution I had for this depends on BuildKit since it treats multi-stage builds as a tree instead of a line. With regular docker builder, this would probably need separate dockerfiles for separate targets. Which is why I wasn't doing it in this PR. I'll put back the pypi install for now.

Reply:
I'm okay removing this for now as long as you leave an issue, especially if it makes the builds faster.

Having a nice docker hub is theoretical, having a faster  build is very much a problem right now.